core/src/com/unciv/logic/battle/Battle.kt

```diff
@@ -152,7 +152,21 @@ object Battle {
         if (defender.getCivInfo().isBarbarian && attackedTile.improvement == Constants.barbarianEncampment)
             defender.getCivInfo().gameInfo.barbarians.campAttacked(attackedTile.position)
 
-        // This needs to come BEFORE the move-to-tile, because if we haven't conquered it we can't move there =)
+        // Must come before normal conquest logic so units that cannot capture cities can still destroy them
+        // Melee units can capture capitals; any unit with CanDestroyCities can destroy non-capitals
+        if (defender.isDefeated() && defender is CityCombatant && attacker is MapUnitCombatant) {
+            if (!defender.city.isCapital()) {
+                val destroyFilters = attacker.unit.getMatchingUniques(UniqueType.CanDestroyCities).flatMap { unique ->
+                    unique.params[0]?.let { sequenceOf(it) } ?: emptySequence()
```

Comment:
Param[0] must exist, or else it wouldn't match the unique, so this should just be .map { unique.params[0] }